tests/ast/test_pre_parser.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from vyper.ast.pre_parser import validate_version_pragma
+from vyper.exceptions import VersionException
+
+
+def test_validate_version_pragma_valid(monkeypatch):
+    _mock_version(monkeypatch)
+    validate_version_pragma(' @version 0.1.0b17', (1, 0))
+    validate_version_pragma(' @version 0.1.3', (1, 0))
+    assert True
+
+
+def test_validate_version_pragma_incompatible(monkeypatch):
+    _mock_version(monkeypatch)
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 0.2.0b17', (1, 0))
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 3.1.0b17', (1, 0))
+
+
+def test_validate_version_pragma_invalid(monkeypatch):
+    _mock_version(monkeypatch)
+    with pytest.raises(VersionException):
+        validate_version_pragma(' @version 0.1', (1, 0))
```

Comment:
Parametrize here too:
```suggestion
@pytest.mark.parametrize('version', [
    '0.2.0b17',
    '3.1.0b17',
])
def test_invalid_version_pragma(monkeypatch, version):
    _mock_version(monkeypatch)
    with pytest.raises(VersionException):
        validate_version_pragma(' @version {version}', (1, 0))
```

Reply:
Yes, Sir!